src/gitops/components/appset/AppSetNavPage.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { ApplicationSetKind, ApplicationSetModel } from '../../models/ApplicationSetModel';
+import {
+  Spinner,
+  Bullseye,
+  Tabs,
+  Tab,
+  TabTitleText,
+} from '@patternfly/react-core';
+import { useApplicationSetActionsProvider } from '../../hooks/useApplicationSetActionsProvider';
+import ResourceDetailsTitle from '../../utils/components/DetailsPageTitle/ResourceDetailsTitle';
+import AppSetDetailsTab from './AppSetDetailsTab';
+import GeneratorsTab from './GeneratorsTab';
+import AppsTab from './AppsTab';
+import EventsTab from './EventsTab';
+import YAMLTab from './YAMLTab';
+import './AppSetNavPage.scss';
+import { useLocation } from 'react-router-dom-v5-compat';
+
+type AppSetPageProps = {
+  name: string;
+  namespace: string;
+  kind: string;
+};
+
+const AppSetNavPage: React.FC<AppSetPageProps> = ({ name, namespace, kind }) => {
+  const location = useLocation();
+  const [activeTabKey, setActiveTabKey] = React.useState<string | number>(0);
+
+  const [appSet, loaded, loadError] = useK8sWatchResource<ApplicationSetKind>({
+    groupVersionKind: {
+      group: 'argoproj.io',
+      version: 'v1alpha1',
+      kind: 'ApplicationSet',
+    },
+    name,
+    namespace,
+  });
+
+  const [actions] = useApplicationSetActionsProvider(appSet);
+
+  // Handle tab query parameter
+  React.useEffect(() => {
+    const searchParams = new URLSearchParams(location.search);
+    const tabParam = searchParams.get('tab');
+    if (tabParam === 'yaml') {
+      setActiveTabKey(1); // YAML tab is at index 1
+    }
+  }, [location.search]);
+
+  if (loadError) return <div>Error loading ApplicationSet details.</div>;
+  if (!loaded || !appSet) return (
+    <Bullseye>
+      <Spinner size="xl" />
+    </Bullseye>
+  );
+
+  const handleTabClick = (
+    event: React.MouseEvent<HTMLElement, MouseEvent>,
+    tabIndex: string | number,
+  ) => {
+    setActiveTabKey(tabIndex);
+  };
+
+  return (
+    <div className="application-set-details-page__main-section">
+      <ResourceDetailsTitle
+        obj={appSet}
+        model={ApplicationSetModel}
+        name={name}
+        namespace={namespace}
+        actions={actions}
+        iconText="AS"
+        iconTitle="Argo CD ApplicationSet"
+        resourcePrefix="Argo CD"
+        showDevPreviewBadge={true}
+      />
+      <div className="application-set-details-page__body">
```

Comment:
I believe in this case, we shouldn't use PF. we should use <HorizontalNav> from @openshift-console/dynamic-plugin-sdk 

If you currently click on the various tabs, you should be able to press the browser's Back and Forward buttons to navigate to the tabs again.